android/sdk/src/main/java/com/taobao/weex/ui/WXRenderManager.java

```diff
@@ -81,6 +91,58 @@ public WXSDKInstance getWXSDKInstance(String instanceId) {
     return statement.getWXSDKInstance();
   }
 
+    public static int getOpenGLRenderLimitValue() {
+      if(mOpenGLRenderLimitValue == 0){
+          int maxsize = 0;
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+              try {
+                  EGLDisplay dpy = EGL14.eglGetDisplay(EGL14.EGL_DEFAULT_DISPLAY);
+                  int[] vers = new int[2];
+                  EGL14.eglInitialize(dpy, vers, 0, vers, 1);
+                  int[] configAttr = {
+                          EGL14.EGL_BUFFER_SIZE, 32,
+                          EGL14.EGL_ALPHA_SIZE, 8,
+                          EGL14.EGL_BLUE_SIZE, 8,
+                          EGL14.EGL_GREEN_SIZE, 8,
+                          EGL14.EGL_RED_SIZE, 8,
+                          EGL14.EGL_RENDERABLE_TYPE, EGL14.EGL_OPENGL_ES2_BIT,
+                          EGL14.EGL_SURFACE_TYPE, EGL14.EGL_WINDOW_BIT,
+                          EGL14.EGL_NONE};
+                  EGLConfig[] configs = new EGLConfig[1];
+                  int[] numConfig = new int[1];
+                  EGL14.eglChooseConfig(dpy, configAttr, 0, configs, 0, configs.length, numConfig, 0);
+                  EGLConfig config = configs[0];
+                  int[] surfAttr = {
+                          EGL14.EGL_WIDTH, 64,
+                          EGL14.EGL_HEIGHT, 64,
+                          EGL14.EGL_NONE};
+                  EGLSurface surf = EGL14.eglCreatePbufferSurface(dpy, config, surfAttr, 0);
+                  int[] ctxAttrib = {
+                          EGL14.EGL_CONTEXT_CLIENT_VERSION, 1,
+                          EGL14.EGL_NONE};
+                  EGLContext ctx = EGL14.eglCreateContext(dpy, config, EGL14.EGL_NO_CONTEXT, ctxAttrib, 0);
+                  EGL14.eglMakeCurrent(dpy, surf, surf, ctx);
+                  int[] maxSize = new int[1];
+                  GLES10.glGetIntegerv(GLES10.GL_MAX_TEXTURE_SIZE, maxSize, 0);
```

Comment:
If there is something unexpected happens, `maxSize[0]` will be 0 as well, which will remove all `LAYER_TYPE_HARDWARE`